typing/datarepr.ml

```diff
@@ -315,3 +317,29 @@ let unboxed_labels_of_type ty_path decl =
         labels rep decl.type_private
   | Type_record _
   | Type_variant _ | Type_abstract _ | Type_open -> []
+
+let constructor_unbound_type_vars_excluding_row_variables cstr =
+  match cstr.cd_res with
+  | None -> TypeSet.empty
+  | Some res_ty ->
+    let tyl = tys_of_constr_args cstr.cd_args in
+    let bound_vars =
+      match get_desc res_ty with
+      | Tconstr (_, args, _) ->
+        args
+        |> List.to_seq
+        |> Seq.filter is_Tvar
+        |> Seq.map Transient_expr.repr
+        |> TypeSet.of_seq
+      | _ -> Misc.fatal_error "cd_res must be Tconstr"
+    in
+    let arg_vars_set =
+      free_vars
+        ~include_row_variables:false
+        (newgenty (Ttuple (List.map (fun ty -> None, ty) tyl)))
```

Comment:
Tupling here is, I believe, more efficient than mapping and set-unioning. Might be nice to have a comment explaining why do this (because, to me, mapping and set-unioning is a more direct statement of what you're trying to do).